dag/src/lib.rs

```diff
@@ -40,6 +41,7 @@ impl<'a> Tree<'a> {
         let offset = dag.get_entry().unwrap().in_number;
         let path = dag.get_entry().unwrap().label.clone();
         let constraints = root.constraints.clone();
+        let custom_gates_constraints = root.custom_gates_constraints.clone();
```

Comment:
We need to store the relation between signals and instantiated custom gate's signals inside a template to apply the appropriate offset to the signal's numbering when traversing the tree